### PR TITLE
Add TiDB best practices on public cloud

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/13564, https://github.com/pingcap/docs/pull/18432

- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices-on-public-cloud.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+在 TiKV 中，[Raft Engine](/glossary.md#raft-engine) 扮演着类似传统数据库中预写日志（WAL）的关键角色。为了达到最佳性能和稳定性，在公有云上部署 TiDB 时，为 Raft Engine 分配专用磁盘至关重要。以下 `iostat` 显示了写密集型工作负载下 TiKV 节点的 I/O 特征。
+
+```
+Device            r/s     rkB/s       w/s     wkB/s      f/s  aqu-sz  %util
+sdb           1649.00 209030.67   1293.33 304644.00    13.33    5.09  48.37
+sdd           1033.00   4132.00   1141.33  31685.33   571.00    0.94 100.00
+```
+
+设备 `sdb` 用于 KV RocksDB，而 `sdd` 用于存储 Raft Engine 日志。注意，`sdd` 的 `f/s` 值明显更高，这表示该设备每秒完成的刷新请求数量更高。在 Raft Engine 中，当 batch 中的写入操作被标记为同步时，batch leader 将在写入后调用 `fdatasync()`，以确保缓存的数据被写入存储。通过为 Raft Engine 分配专用磁盘，TiKV 可以减少请求的平均队列长度，从而确保写入延迟的最优化和稳定性。
```
Comment (qiancai) [inline best-practices-on-public-cloud.md:55]: "当 batch 中的写入操作被标记为同步时" 不确定对不对，英文为 “when a write in a batch is marked synchronous”

File: best-practices-on-public-cloud.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+要减少 TiKV 实例中跨可用区的写入流量，你可以启用 gRPC 压缩功能，该功能在网络传输数据之前会对其进行压缩。以下配置示例展示了如何为 TiKV 启用 gzip gRPC 压缩：
+
+```
+server_configs:
+  tikv:
+    server.grpc-compression-type: gzip
+```
+
+要减少 TiFlash MPP 任务中数据交换（data shuffle 过程）所带来的网络流量，建议在同一可用区内部署多个 TiFlash 实例。从 v6.6.0 开始，[Exchange 数据压缩](/explain-mpp.md#mpp-version-和-exchange-数据压缩)功能默认启用，以减少了 MPP 数据交换导致的网络流量。
```
Comment (qiancai) [inline best-practices-on-public-cloud.md:147]:
```suggestion
要减少 TiFlash MPP 任务中数据交换（data shuffle 过程）所带来的网络流量，建议在同一可用区内部署多个 TiFlash 实例。从 v6.6.0 开始，[Exchange 数据压缩](/explain-mpp.md#mpp-version-和-exchange-数据压缩)功能默认启用，以减少 MPP 数据交换导致的网络流量。
```

File: best-practices-on-public-cloud.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+# 在公有云上部署 TiDB 的最佳实践
+
+随着公有云基础设施的普及，越来越多的用户选择在公有云上部署和管理 TiDB。然而，要想在公有云上充分发挥 TiDB 的性能，需要关注多个方面，包括性能优化、成本控制、系统可靠性和可扩展性。
+
+本文介绍在公有云上部署 TiDB 的一系列最佳实践，例如减少 KV RocksDB 中的压缩 I/O 流量、为 Raft Engine 配置专用磁盘、优化跨可用区的流量成本、缓解 Google Cloud 上实时迁移维护事件带来的性能影响，以及对大规模集群中的 PD Server 进行性能调优。遵循这些最佳实践，可以显著提升 TiDB 在公有云上的性能、成本效率、可靠性和可扩展性。
+
+## 减少 KV RocksDB 中的压缩 I/O 流量
+
+[RocksDB](https://rocksdb.org/) 是 TiKV 的存储引擎，负责存储用户数据。出于成本考虑，云上提供的 EBS IO 吞吐量通常比较有限，因此 RocksDB 可能会表现出较高的写放大，导致磁盘吞吐量成为负载的瓶颈。随着时间的推移，待压缩的字节总量会不断增加从而触发流量控制，这意味着此时 TiKV 缺乏足够的磁盘带宽来处理前台写入流量。
```
Comment (qiancai) [inline best-practices-on-public-cloud.md:14]:
```suggestion
## 减少 KV RocksDB 中的 compaction I/O 流量

[RocksDB](https://rocksdb.org/) 是 TiKV 的存储引擎，负责存储用户数据。出于成本考虑，云上提供的 EBS IO 吞吐量通常比较有限，因此 RocksDB 可能会表现出较高的写放大，导致磁盘吞吐量成为负载的瓶颈。随着时间的推移，待 compaction 的字节总量会不断增加从而触发流量控制，这意味着此时 TiKV 缺乏足够的磁盘带宽来处理前台写入流量。
```

File: best-practices-on-public-cloud.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+### 优化 PD 性能
+
+为了解决 PD Server 的高 CPU 利用率问题，可以进行以下调整：
+
+#### 调整 PD 配置
+
+[`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval)：此参数控制 PD Server 更新物理 TSO batch 的间隔。通过缩短此间隔，PD Server 可以更频繁地分配 TSO batch，从而减少下一次分配的等待时间。
+
+```
+tso-update-physical-interval = "10ms" # 默认值: 50ms
```
Comment (qiancai) [inline best-practices-on-public-cloud.md:183]:
```suggestion
tso-update-physical-interval = "10ms" # 默认值为 50ms
```

File: best-practices-on-public-cloud.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
+---
+title: 在公有云上部署 TiDB 的最佳实践
+summary: 了解在公有云上部署 TiDB 的最佳实践。
+---
+
+# 在公有云上部署 TiDB 的最佳实践
+
+随着公有云基础设施的普及，越来越多的用户选择在公有云上部署和管理 TiDB。然而，要想在公有云上充分发挥 TiDB 的性能，需要关注多个方面，包括性能优化、成本控制、系统可靠性和可扩展性。
+
+本文介绍在公有云上部署 TiDB 的一系列最佳实践，例如减少 KV RocksDB 中的压缩 I/O 流量、为 Raft Engine 配置专用磁盘、优化跨可用区的流量成本、缓解 Google Cloud 上实时迁移维护事件带来的性能影响，以及对大规模集群中的 PD Server 进行性能调优。遵循这些最佳实践，可以显著提升 TiDB 在公有云上的性能、成本效率、可靠性和可扩展性。
```
[the remark this answers is not in the document]
Comment (dbsid) [reply best-practices-on-public-cloud.md:10]: lgtm

File: best-practices-on-public-cloud.md (hunk 3 above)
Comment (dbsid) [reply best-practices-on-public-cloud.md:14]: lgtm

**Review by dbsid — APPROVED** [review]

File: best-practices-on-public-cloud.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+#### 高端磁盘
+
+如果你希望 Raft Engine 有更低的延迟，可以考虑使用高端磁盘。对于不同的公有云，推荐的高端磁盘如下：
+
+- AWS：推荐使用 [io2](https://docs.aws.amazon.com/zh_cn/ebs/latest/userguide/ebs-volume-types.html)。磁盘大小和 IOPS 可以根据你的具体需求进行配置。
+
+- Google Cloud：推荐使用 [pd-extreme](https://cloud.google.com/compute/docs/disks?hl=zh-cn#disk-types/)。磁盘大小、IOPS 和 MBPS 可以进行配置，但仅在 CPU 核数超过 64 的实例上可用。
+
+- Azure：推荐使用 [ultra disk](https://learn.microsoft.com/zh-cn/azure/virtual-machines/disks-types#ultra-disks)。磁盘大小、IOPS 和 MBPS 可以根据你的具体需求进行配置。
```
Comment (hfxsd) [inline best-practices-on-public-cloud.md:81]:
```suggestion
- Azure：推荐使用 [Ultra Disk](https://learn.microsoft.com/zh-cn/azure/virtual-machines/disks-types#ultra-disks)。磁盘大小、IOPS 和 MBPS 可以根据你的具体需求进行配置。
```

File: best-practices-on-public-cloud.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+server_configs:
+  tikv:
+    server.grpc-compression-type: gzip
+```
+
+要减少 TiFlash MPP 任务中数据交换（data shuffle 过程）所带来的网络流量，建议在同一可用区内部署多个 TiFlash 实例。从 v6.6.0 开始，[Exchange 数据压缩](/explain-mpp.md#mpp-version-和-exchange-数据压缩)功能默认启用，以减少 MPP 数据交换导致的网络流量。
+
+## 缓解 Google Cloud 上的实时迁移维护事件带来的性能影响
+
+Google Cloud 的 [实时迁移功能](https://cloud.google.com/compute/docs/instances/live-migration-process?hl=zh-cn)允许虚拟机在主机之间无缝迁移而不会导致停机。然而，这些迁移事件虽不频繁，但可能会显著影响虚拟机的性能，其中包括那些在 TiDB 集群中运行的虚拟机。在这些事件发生期间，受影响的虚拟机可能会出现性能下降，导致 TiDB 集群中的查询处理时间增加。
```
Comment (hfxsd) [inline best-practices-on-public-cloud.md:151]:
```suggestion
Google Cloud 的[实时迁移功能](https://cloud.google.com/compute/docs/instances/live-migration-process?hl=zh-cn)允许虚拟机在主机之间无缝迁移而不会导致停机。然而，这些迁移事件虽不频繁，但可能会显著影响虚拟机的性能，其中包括那些在 TiDB 集群中运行的虚拟机。在这些事件发生期间，受影响的虚拟机可能会出现性能下降，导致 TiDB 集群中的查询处理时间增加。
```

File: best-practices-on-public-cloud.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+通过使用该监控脚本，并在维护事件期间采取必要的措施，TiDB 集群可以更好地应对 Google Cloud 上的实时迁移事件，确保对查询处理和响应时间的影响最小以及系统的平稳运行。
+
+## 为具有高 QPS 的大规模 TiDB 集群优化 PD 性能
+
+在 TiDB 集群中，一个活跃的 Placement Driver (PD) Server 承担着许多关键任务，例如处理提供 TSO (Timestamp Oracle) 和处理请求。然而，依赖单个活跃 PD Server 可能会限制 TiDB 集群的扩展性。
+
+### PD 限制的特征
+
+以下图表展示了一个由三个 PD Server 组成的大规模 TiDB 集群的特征，其中每个 PD Server 均配置了 56 核的 CPU。可以看出，当每秒查询数（QPS）超过 100 万次且每秒 TSO 请求数超过 162,000 次时，CPU 利用率达到约 4,600%。这一高 CPU 利用率表明 PD Leader 的负载已经相当高且可用的 CPU 资源即将耗尽。
```
Comment (hfxsd) [inline best-practices-on-public-cloud.md:169]:
```suggestion
以下图表展示了一个由三个 PD Server 组成的大规模 TiDB 集群的特征，其中每个 PD Server 均配置了 56 核的 CPU。可以看出，当每秒查询数（QPS）超过 100 万次且每秒 TSO 请求数超过 162,000 次时，CPU 利用率达到约 4600%。这一高 CPU 利用率表明 PD Leader 的负载已经相当高且可用的 CPU 资源即将耗尽。
```

File: best-practices-on-public-cloud.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+[coprocessor]
+  region-split-size = "256MB"
+```
+
+## 调整后的效果
+
+调整后的效果如下：
+
+- 每秒 TSO 请求数减少到 64,800 次。
+- CPU 利用率显著降低，从约 4,600% 降低到 1,400%。
```
Comment (hfxsd) [inline best-practices-on-public-cloud.md:208]:
```suggestion
- CPU 利用率显著降低，从约 4600% 降低到 1400%。
```

**Review by hfxsd — APPROVED** [review]

File: best-practices-on-public-cloud.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+  region-split-size = "256MB"
+```
+
+## 调整后的效果
+
+调整后的效果如下：
+
+- 每秒 TSO 请求数减少到 64,800 次。
+- CPU 利用率显著降低，从约 4600% 降低到 1400%。
+- `PD server TSO handle time` 的 P999 值从 2ms 降低到 0.5ms。
```
Comment (qiancai) [inline best-practices-on-public-cloud.md:209]:
```suggestion
- `PD server TSO handle time` 的 P999 值从 2 ms 降低到 0.5 ms。
```

File: best-practices-on-public-cloud.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+
+随着公有云基础设施的普及，越来越多的用户选择在公有云上部署和管理 TiDB。然而，要想在公有云上充分发挥 TiDB 的性能，需要关注多个方面，包括性能优化、成本控制、系统可靠性和可扩展性。
+
+本文介绍在公有云上部署 TiDB 的一系列最佳实践，例如减少 KV RocksDB 中的 compaction I/O 流量、为 Raft Engine 配置专用磁盘、优化跨可用区的流量成本、缓解 Google Cloud 上实时迁移维护事件带来的性能影响，以及对大规模集群中的 PD Server 进行性能调优。遵循这些最佳实践，可以显著提升 TiDB 在公有云上的性能、成本效率、可靠性和可扩展性。
+
+## 减少 KV RocksDB 中的 compaction I/O 流量
+
+[RocksDB](https://rocksdb.org/) 是 TiKV 的存储引擎，负责存储用户数据。出于成本考虑，云上提供的 EBS IO 吞吐量通常比较有限，因此 RocksDB 可能会表现出较高的写放大，导致磁盘吞吐量成为负载的瓶颈。随着时间的推移，待 compaction 的字节总量会不断增加从而触发流量控制，这意味着此时 TiKV 缺乏足够的磁盘带宽来处理前台写入流量。
+
+要缓解由磁盘吞吐量受限引起的瓶颈，你可以通过[启用 Titan](#enable-titan) 来改善性能。如果数据的平均行大小低于 512 字节，Titan 并不适用，此时你可以通过[提高所有压缩级别](#increase-all-the-compression-levels) 来改善性能。
```
Comment (qiancai) [inline best-practices-on-public-cloud.md:16]:
```suggestion
要缓解由磁盘吞吐量受限引起的瓶颈，你可以通过[启用 Titan](#启用-titan) 来改善性能。如果数据的平均行大小低于 512 字节，Titan 并不适用，此时你可以通过[提高所有压缩级别](#提高所有压缩级别) 来改善性能。
```

**Pull request merged**